orttraining/orttraining/python/training/ortmodule/ortmodule.py

```diff
@@ -322,3 +305,25 @@ def __setattr__(self, name: str, value) -> None:
         else:
             # Setting any new attributes should be done on ORTModule only when 'torch_module' is not defined
             self.__dict__[name] = value
+
+    def __getstate__(self):
+        # Attempting to serialize ORTModule
+
+        state = _utils.get_state_after_deletion_of_non_ortmodule_methods(self, self.module)
+        return state
+
+    def __setstate__(self, state):
+        # Attempt to deserialize ORTModule
+
+        self.__dict__.update(state)
+
+        # Re-register contrib OPs
+        pytorch_export_contrib_ops.register()
+        CustomOpSymbolicRegistry.register_all()
+        CustomGradientRegistry.register_all()
+
+        # Re-initialize the ORTModule forward method
+        _utils.patch_ortmodule_forward_method(self)
+
+        # Re-bind users custom methods to ORTModule
+        _utils.check_for_name_collisions_and_bind_methods_to_ortmodule(self, self.module)
```

Comment:
Is it possible to combine all sorts of reinitialization into a single method that does it all?

Having to implement getstate/setstate in several classes is hard enough (ortmodule, graph execution maanger, training manager and inference manager). Adding calling method by method in each of them makes things even harder to maintain